src/Cell.js

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+import Colours from "./Colours";
+
+export default class Cell extends React.Component {
+    render() {
+        return (
+            <button
+                className={this.props.value === 0 ? "cell empty" : "cell"}
+                ref={this.props.cellRef} row={this.props.row}
+                col={this.props.col} onClick={()=>{ this.props.handleClick(this.props)}}
+                onKeyDown={(event) => { this.props.handleKeyDown(event, this.props)}}
```

Comment:
For C2LC, we have been following the naming convention of on* for public component API event emitters, and handle* for internal event handlers. I made a quick commit to illustrate:

https://github.com/simonbates/react-launchpad-sliding-puzzle/commit/228e46b2e6edc750ee210eaeb376f419d9520136

This commit includes the following additional changes:

- "public class field syntax" for event handlers (`handleClick = (event) => {...};`), which enables binding of an event handler to the component `this`, without the need to explicitly use `bind` or an anonymous function within the `render` method
- I changed the API of the Cell `onClick` and `onKeyDown` to provide the row and column directly as callback parameters, to lessen the knowledge that clients of Cell need to have about its implementation (so we could change the mechanism that Cell uses to store its row and col, and clients only need to depend on the public API)
- I use `this.props.row` and `this.props.col` directly and do not store them in the DOM (for instances where we want to store data in the DOM, we have been using the HTML data attribute support, for example: https://github.com/codelearncreate/c2lc-coding-environment/blob/e632e93e8549a4fe814dd0440f005bba946a76f2/src/ProgramBlockEditor.js#L265)

Reply:
Another quick thought on `handleClick(row, col)` vs `handleClick(cellProps)`. Very much related to what I wrote above, but maybe worth writing explicitly. Something I like about `handleClick(row, col)` is that if `row` and `col` are well known types, such as numbers in this case, it reduces the amount of code that one needs to look at to understand the method. For `handleClick(cellProps)`, one needs to know what the type of `cellProps` is and where it is defined, and then you need to look at it to understand it. This may be more or less work, depending on the programming language and tooling, but even in the best case (the type is clearly and easily available), it's still more to read. And the type may have members that aren't relevant to the method at hand.